### PR TITLE
DOCS: more usage examples, new API section, other minor updates

### PR DESCRIPTION
**Pull request by kylezeeuwen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
...
 
+## Rosie API
+
+As stated above the rosie factory signatures can be broken into factory definition and object creation.
+
+Additionally factories can be defined and accessed via the Factory singleton, or they can be created and maintained by the callee.
+
+###Factory declaration functions
+
+Once you have an instance returned from a `Factory.define` call, you do the actual of work of defining the objects. This is done the methods below on the Factory instance that is returned from `Factory.define` or from `new Factory()`  (note these are typically chained together as in the examples above):
```
Comment (eventualbuddha) [inline README.md]: "This is done *in*"

**Comment by kylezeeuwen** [comment]
Good catch typo fixed and I cleaned up that sentence a bit. 

Once all feedback is processed, do you prefer that I squash my changes into a single commit ? If so let me know.

File: README.md [hunk 2]
```diff
@@ -39,33 +46,69 @@ Factory.define('player')
...
+  .attr('seasonStart', '2016-01-01')
+  .option('numMatches', 2)
+  .attr('matches', ['numMatches', 'seasonStart'], function(numMatches, seasonStart) {
+    var matches = [];
+    for (var i = 1; i <= numMatches; i++) {
+      matches.push({
+        matchDate: moment(seasonStart).add(i, 'week').format('YYYY-MM-DD'),
+        homeScore: Math.floor(Math.random() * 5),
+        awayScore: Math.floor(Math.random() * 5)
+      })
```
Comment (eventualbuddha) [inline README.md]: Let's use semicolons consistently in the examples.

File: README.md [hunk 3]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
 
 A tool like [babel](https://babeljs.io) is currently required to use this syntax.
 
+## Rosie API
+
+As stated above the rosie factory signatures can be broken into factory definition and object creation.
+
+Additionally factories can be defined and accessed via the Factory singleton, or they can be created and maintained by the callee.
+
+###Factory declaration functions
```
Comment (eventualbuddha) [inline README.md]: Space after hashes.

File: README.md [hunk 4]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
...
+* **Factory.define(`factory_name`, `constructor`)** - Optionally pass a constuctor function, and the objects produced by `.build` will be passed through the `constructor` funtion.
+
+
+#### instance.attr:
+
+Use this to define attributes of your objects
+
+* **instance.attr(`attribute_name`, `default_value`)** - `attribute_name` is required and is a string, `default_value` is the value to use by default for the attribute
+* **instance.attr(`attribute_name`, `generator_function`)** - `generator_function` is called to generate the value of the attribute
+* **instance.attr(`attribute_name`, `dependencies`, `generator_function`)** - `dependencies` is an array of strings, each string is the name of an attribute or option that is required by the `generator_function` to generate the value of the attribute. This string of `dependencies` will match the parameters that are passed to the `generator_function`
```
Comment (eventualbuddha) [inline README.md]: "This _list_ of dependencies…" using the word "string" to mean "list" is confusing in this context.

File: README.md [hunk 5]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
...
+
+#### instance.option:
+
+Use this to define options used to programatically generate attributes of your objects
```
Comment (eventualbuddha) [inline README.md]: This should say something about configuring attributes rather than generating them.

File: README.md [hunk 6]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
...
+* **instance.option(option_name, `default_value`)** - option_name is required and is a string, `default_value` is the value to use by default for the option
+* **instance.option(option_name, `generator_function`)** - `generator_function` is called to generate the value of the option
+* **instance.option(option_name, `dependencies`, `generator_function`)** - `dependencies` is an array of strings, each string is the name of an attribute or option that is required by the `generator_function` to generate the value of the option. This string of `dependencies` will match the parameters that are passed to the `generator_function`
+
+#### instance.sequence:
+
+Use this to define an auto incrementing sequence field in your object
+
+* **instance.sequence(`sequence_name`)** - define a sequence called `sequence_name`, set the start value to 1
+* **instance.sequence(`sequence_name`, `generator_function`)** - `generator_function` is called to generate the value of the sequence
```
Comment (eventualbuddha) [inline README.md]: This should indicate that the generator function may take the sequence numbers as an argument.

File: README.md [hunk 7]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
...
+* **instance.sequence(`sequence_name`, `dependencies`, `generator_function`)** - `dependencies` is an array of strings, each string is the name of an attribute or option that is required by the `generator_function` to generate the value of the option. This string of `dependencies` will match the parameters that are passed to the `generator_function`
```
Comment (eventualbuddha) [inline README.md]: This is true, except that the first argument will not be a dependency, but the sequence number.

File: README.md [hunk 8]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
...
+
+#### instance.after:
+
+* **instance.after(`callback`)** - register a `callback` function that will be called at the end of the object build process. The `callback` is invoked with two params: (`build_object`, `object_options`)
+
+###Object building functions
```
Comment (eventualbuddha) [inline README.md]: Space after hashes.

File: README.md [hunk 9]
```diff
@@ -135,6 +222,66 @@ const game = Game.build({is_over: true});
...
+
+* **instance.attr(`attribute_name`, `default_value`)** - `attribute_name` is required and is a string, `default_value` is the value to use by default for the attribute
+* **instance.attr(`attribute_name`, `generator_function`)** - `generator_function` is called to generate the value of the attribute
+* **instance.attr(`attribute_name`, `dependencies`, `generator_function`)** - `dependencies` is an array of strings, each string is the name of an attribute or option that is required by the `generator_function` to generate the value of the attribute. This string of `dependencies` will match the parameters that are passed to the `generator_function`
+
+#### instance.option:
+
+Use this to define options used to programatically generate attributes of your objects
+
+* **instance.option(option_name, `default_value`)** - option_name is required and is a string, `default_value` is the value to use by default for the option
```
Comment (eventualbuddha) [inline README.md]: Why doesn't option name have ticks?

**Comment by kylezeeuwen** [comment]
All feedback addressed and pushed. Specific changes:

* semicolon consistency
* spaces after hashes
* string -> list in "This string of dependencies"
* sequence generator function docs updated to account for param1=++sequence
* Description of intance.options updated
* Some section links added in API section pointing back to examples
* option_name -> `option_name`

**Comment by eventualbuddha** [comment]
Thanks!

**Pull request merged**